test-suite/Djinni.cmake

```diff
@@ -1,17 +1,20 @@
-
-# Resolve DJINNI_EXECUTABLE
-set(DJINNI_EXECUTABLE_NAMES "djinni")
-find_program(DJINNI_EXECUTABLE ${DJINNI_EXECUTABLE_NAMES}
-  DOC "Path to the Djinni executable"
-  PATHS ${DJINNI_EXECUTABLE_PATH})
-if(DJINNI_EXECUTABLE)
-  execute_process(COMMAND ${DJINNI_EXECUTABLE} "--version" OUTPUT_VARIABLE DJINNI_VERSION)
-  string(REGEX REPLACE "\n+$" "" DJINNI_VERSION "${DJINNI_VERSION}")
-  message(STATUS "Found Djinni: ${DJINNI_EXECUTABLE} (${DJINNI_VERSION})")
+cmake_minimum_required(VERSION 3.18)
+
+# find Djinni executable.
+# On windows find_program() does not work for finding the `djinni.bat` script.
+# The script must either be on the PATH or `DJINNI_EXECUTABLE` must explicitly be predefined.
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+  if(NOT DEFINED CACHE{DJINNI_EXECUTABLE})
+    set(DJINNI_EXECUTABLE djinni.bat CACHE FILEPATH "path of djinni binary")
+  endif()
 else()
-  message(FATAL_ERROR "Could not find DJINNI_EXECUTABLE using the following names: ${DJINNI_EXECUTABLE_NAMES}")
+  find_program(DJINNI_EXECUTABLE djinni REQUIRED)
 endif()
 
+execute_process(COMMAND ${DJINNI_EXECUTABLE} "--version" OUTPUT_VARIABLE DJINNI_VERSION)
+string(REGEX REPLACE "\n+$" "" DJINNI_VERSION "${DJINNI_VERSION}")
+message(STATUS "Found Djinni: ${DJINNI_EXECUTABLE} (${DJINNI_VERSION})")
```

Comment:
We might end up with a duplicate message now, the first one being printed by `find_program`. Not sure how to fix it, maybe it's not something worth worrying about just yet.

Reply:
I have looked into the workflow logs & I could not find any duplicate output. Be aware that I moved (and not copied) this section out of the find_program part. That way it will be executed for both windows (which does not work with find_program) & the rest.